model/channel_search.go

```diff
@@ -14,6 +14,9 @@ type ChannelSearch struct {
 	Term                   string `json:"term"`
 	ExcludeDefaultChannels bool   `json:"exclude_default_channels"`
 	NotAssociatedToGroup   string `json:"not_associated_to_group"`
+	Paginate               bool   `json:"paginate"`
+	Page                   int    `json:"page"`
+	PerPage                int    `json:"per_page"`
```

Comment:
What about using pointers for one or both of `Page` and `PerPage`? Then if `Page` is nil we don't do any paging. As an API user I would probably just provider `Page` and `PerPage` the first time without reading the docs or realizing that I need to set `Paginate` to `true`. Especially since all our other endpoints just require me to set page and per_page in the query string, not anything else

Reply:
Yes I like this idea. I'll make this change.

Reply:
Update committed.